pkg/crc/preflight/preflight_daemon_task_check_windows.go

```diff
@@ -0,0 +1,138 @@
+package preflight
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/code-ready/crc/pkg/os/windows/powershell"
+)
+
+var (
+	// https://docs.microsoft.com/en-us/windows/win32/taskschd/daily-trigger-example--xml-
+	daemonTaskTemplate = `<?xml version="1.0" encoding="UTF-16"?>
+<Task version="1.3" xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task">
+  <RegistrationInfo>
+    <Description>Run crc daemon as a task</Description>
+    <Version>%s</Version>
+  </RegistrationInfo>
+  <Settings>
+    <DisallowStartIfOnBatteries>false</DisallowStartIfOnBatteries>
+    <StopIfGoingOnBatteries>false</StopIfGoingOnBatteries>
+    <Hidden>true</Hidden>
+    <MultipleInstancesPolicy>IgnoreNew</MultipleInstancesPolicy>
+    <IdleSettings>
+      <Duration>PT10M</Duration>
+      <WaitTimeout>PT1H</WaitTimeout>
+      <StopOnIdleEnd>true</StopOnIdleEnd>
+      <RestartOnIdle>false</RestartOnIdle>
+    </IdleSettings>
+    <UseUnifiedSchedulingEngine>true</UseUnifiedSchedulingEngine>
+  </Settings>
+  <Triggers />
+  <Actions Context="Author">
+    <Exec>
+      <Command>powershell.exe</Command>
+      <Arguments>-WindowStyle Hidden -Command %s</Arguments>
+    </Exec>
+  </Actions>
+</Task>
+`
+)
+
+func genDaemonTaskInstallTemplate(crcVersion, daemonServiceName string) string {
+	return fmt.Sprintf(daemonTaskTemplate,
+		crcVersion,
+		daemonServiceName,
+	)
+}
+
+func checkIfDaemonTaskInstalled() error {
+	_, stderr, err := powershell.Execute("Get-ScheduledTask", "-TaskName", constants.DaemonTaskName)
+	if err != nil {
+		logging.Debugf("%s task is not installed: %v : %s", constants.DaemonTaskName, err, stderr)
+		return err
+	}
+	if err := checkIfOlderTask(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func fixDaemonTaskInstalled() error {
+	// prepare the task script
+	binPath, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("unable to find the current executable location: %v", err)
+	}
+	binPathWithArgs := fmt.Sprintf("%s daemon", binPath)
+	var buf bytes.Buffer
+	if err := xml.EscapeText(&buf, []byte(binPathWithArgs)); err != nil {
+		return err
+	}
```

Comment:
This belongs in `genDaemonTaskInstallTemplate`, not here. If you do it here, this means all callers of `genDaemonTaskInstallTemplate` have to be aware of it. This also kind of hardcodes in `fixDaemonTaskInstalled` that `genDaemonTaskInstallTemplate` will be doing `fmt.Sprintf(xml, ...)`. `fixDaemonTaskInstalled` has no need to know that, this just muddles unrelated abstraction layers

Reply:
Not very easy to read with the XML embedded in the function, but I had in mind something like this:
```
func genDaemonTaskInstallTemplate(crcVersion, daemonServiceName string) (string, error) {
       var escapedName bytes.Buffer
	if err := xml.EscapeText(&escapedName, []byte(daemonServiceName)); err != nil {
		return "", err
	}
	return fmt.Sprintf(daemonTaskTemplate,
		crcVersion,
		escapedName.String(),
	), nil
}
```